File: designerPdfViewer.pl
#!/usr/bin/perl
use strict;
use warnings;
use Test::More "no_plan";

# Complete the designerPdfViewer function below.
sub designerPdfViewer {
   my($h, $word) = @_;

   my $max = 0;
   my $aOrd = ord('a');
   foreach my $letter(split '', $word) {
      my $height = $h->[ord($letter) - $aOrd];
      $max = $height if $height > $max;
   }

   return $max * (length $word);
}

# Test cases
is(designerPdfViewer([qw(1 3 1 3 1 4 1 3 2 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5)], "abc"),  9,  "test 1");
is(designerPdfViewer([qw(1 3 1 3 1 4 1 3 2 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 7)], "zaba"), 28, "test 2");

File: designerPdfViewer.py
#!/bin/python3
import unittest

# Complete the designerPdfViewer function below.
def designerPdfViewer(h, word):
    max = 0;
    aOrd = ord('a');
    for letter in word:
        height = int(h[ord(letter) - aOrd])
        if height > max:
            max = height

    return max * len(word)

# Test cases
class DesignerPdfViewerTest(unittest.TestCase):
    def test1(self):
        self.assertEqual(designerPdfViewer("1 3 1 3 1 4 1 3 2 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5".split(), "abc"), 9)
    
    def test2(self):
        self.assertEqual(designerPdfViewer("1 3 1 3 1 4 1 3 2 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 5 7".split(), "zaba"), 28)
    
if __name__ == '__main__':
    unittest.main()
